src/libs/EmojiTrie.js

```diff
@@ -0,0 +1,14 @@
+import emojis from '../../assets/emojis';
+import Trie from './Trie';
+
+// Create a Trie object
+const emojisTrie = new Trie();
+
+// Inserting all emojis into the Trie object
+for (let i = 0; i < emojis.length; i++) {
+    if (emojis[i].name) {
+        emojisTrie.add(emojis[i].name, {code: emojis[i].code});
```

Comment:
I will try to add them.

Reply:
To add the emojis keywords we can consider each keyword as a node in the tree but without  the `code` property.

i.e when adding this emoji 
```javascript   {
        code: '😀',
        'keywords': [
          'face',
          'grin'
        ],
        name: 'grinning'
      }
``` 
instead of adding one node to the tree in this shape 
```javascript
{name: 'grinning', metadata: {code: '😀'}}
```
we should add these 3 nodes 
```javascript
{name: 'grinning', metadata: {code: '😀', suggestions: []}}
{name: 'face', metadata: {suggestions: [{code: '😀', name: 'grinning'}]}}
{name: 'grin', metadata: {suggestions: [{code: '😀', name: 'grinning'}]}}
```

in this case, our tree has two types of nodes :

- node without metadata `code` which represents an emoji keyword.
- node with metadata `code` which represent a emoji.

so when replacing an emoji name with its code we should eliminate any node that doesn't have the `code` property.

Also, we should update `getAllMatchingWords` method to get all nodes started with a substring and their `suggestions`.

I almost finished it but I getting your thoughts if you have a better idea.

Reply:
Finished.

https://user-images.githubusercontent.com/108357004/190903788-6c078504-f32b-4248-8f38-7de70f291d0a.mov